test/forms/TrusteesUkAddressFormProviderSpec.scala

```diff
@@ -75,5 +125,38 @@ class TrusteesUkAddressFormProviderSpec extends StringFieldBehaviours {
       fieldName,
       requiredError = FormError(fieldName, requiredKey)
     )
+
+    behave like nonEmptyField(
+      form,
+      fieldName,
+      requiredError = FormError(fieldName, requiredKey, Seq(fieldName))
+    )
+
+  }
+
+  ".postcode" must {
+
+    val fieldName = "postcode"
+    val requiredKey = "trusteesUkAddress.error.postcode.required"
+
+    behave like fieldThatBindsValidData(
```

Comment:
We have tested this differently elsewhere for postcodes.

```scala
behave like fieldWithRegexpWithGenerator(
      form,
      fieldName,
      regexp = postcodeRegexWithSpaceConstrained,
      generator = arbitrary[String],
      error = FormError(fieldName, invalidKey)
    )
```

Reply:
Refactored to:

    behave like fieldWithRegexpWithGenerator(
      form,
      fieldName,
      regexp = Validation.postcodeRegex,
      generator = arbitrary[String],
      error = FormError(fieldName, invalidKey, Seq(Validation.postcodeRegex))
    )

Reply:
Done